execution-environment/execution-environment.yaml

```diff
@@ -0,0 +1,26 @@
+version: 3
+
+images:
+  base_image:
+    name: registry.redhat.io/ansible-automation-platform/ee-minimal-rhel9:2.18
+
+dependencies:
+  galaxy: requirements.yaml
+  python: requirements.txt
+  ansible_core:
+    package_pip: ansible-core
+  ansible_runner:
+    package_pip: ansible-runner
+
+options:
+  package_manager_path: /usr/bin/microdnf
+
+additional_build_files:
+  - src: openshift-clients.rpm
```

Comment:
An alternative would be to get the clients as a tar from https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable/ which is publicly available.

Reply:
Good plan. I'll make that change.

Reply:
Wait, no, we can't -- at least, not without getting hacky. There is an rpm dependency on the `openshift-clients` package, and the environment will fail to build without it.

Reply:
The build fails like this:

```
+ PKGMGR_OPTS='--nodocs --setopt install_weak_deps=0'
+ mkdir -p /output/bindep
+ mkdir -p /output/wheels
+ mkdir -p /tmp/src
+ cd /tmp/src
+ PACKAGES=
+ PIP_OPTS=
+ install_bindep
+ '[' -f bindep.txt ']'
+ bindep -l newline
+ sort
+ '[' rhel == centos ']'
++ bindep -b compile
++ true
+ compile_packages=openshift-clients
+ '[' '!' -z openshift-clients ']'
+ /usr/bin/microdnf install -y --nodocs --setopt install_weak_deps=0 openshift-clients
error: No package matches 'openshift-clients'
Error: building at STEP "RUN /output/scripts/assemble": while running runtime: exit status 1
```

Let me see if there's any way to make this optional.

Reply:
what is that dependency?

Reply:
I'm not sure, but there's some discussion about it here: https://access.redhat.com/solutions/6985157

Reply:
Ah, that says:

> The collection [redhat.openshift](https://console.redhat.com/ansible/automation-hub/repo/published/redhat/openshift/) has a dependency on the collection [kubernetes.core](https://console.redhat.com/ansible/automation-hub/repo/published/redhat/openshift/dependencies/). One of the requirements listed in the [bindep.txt](https://access.redhat.com/bounce/?externalURL=https%3A%2F%2Fgithub.com%2Fansible-collections%2Fkubernetes.core%2Fblob%2Fmain%2Fbindep.txt) file for installing the kubernetes.core collection is the package openshift-clients, which is a component of the RHOCP repositories.

Reply:
otherwise  with image has the openshift-clients pre-installed: https://catalog.redhat.com/software/containers/ansible-automation-platform-25/ee-supported-rhel9/650a56807ab682340da38ad5?container-tabs=packages&rh-tabs-0yyn0ehxp=rh-tab-tg1692g4p

but it don't know if it makes sense to use it with the ansible builder

Reply:
feel free to choose whatever solution, it looks like there no easy way :cry:

Reply:
I think I'll just leave it as is for now. In the future, maybe we'll add some credentials to a github workflow so that it could download the package and build the image.